pages_project.go

```diff
@@ -67,18 +67,17 @@ type PagesProjectDeploymentConfigs struct {
 
 // PagesProjectDeploymentConfigEnvironment represents the configuration for preview or production deploys.
 type PagesProjectDeploymentConfigEnvironment struct {
-	EnvVars            map[string]PagesProjectDeploymentVar `json:"env_vars"`
-	CompatibilityDate  string                               `json:"compatibility_date,omitempty"`
-	CompatibilityFlags []string                             `json:"compatibility_flags,omitempty"`
-	KvNamespaces       NamespaceBindingMap                  `json:"kv_namespaces,omitempty"`
-	DoNamespaces       NamespaceBindingMap                  `json:"durable_object_namespaces,omitempty"`
-	D1Databases        D1BindingMap                         `json:"d1_databases,omitempty"`
-	R2Bindings         R2BindingMap                         `json:"r2_buckets,omitempty"`
-}
-
-// PagesProjectDeploymentVar represents a deployment environment variable.
-type PagesProjectDeploymentVar struct {
-	Value string `json:"value"`
+	EnvVars                          EnvironmentVariableMap `json:"env_vars,omitempty"`
```

Comment:
in a typed language, this is a bit of a code smell. instead of including the type (map) in the name, could we call it EnvVars that expects a `[]map[string]EnvironmentVariable`?

Reply:
We use these types internally, so I'd personally like to keep this consistent. I can change if you really want though

Reply:
alright, let's keep them for consistency but the only caveat is once we start looking at codegen'ing the library, this will be changed again to be the concrete type of `map[string]*EnvironmentVariable` so we'll be inconsistent there once that happens.